### PR TITLE
pybind/mgr: Fix IPv6 url generation

### PR DESCRIPTION
**Pull request by sebastian-philipp**
Fixes: https://tracker.ceph.com/issues/52117
Signed-off-by: Sebastian Wagner <sewagner@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/dashboard/module.py [hunk 1]
```diff
@@ -33,7 +33,7 @@
 from .services.sso import SSO_COMMANDS, handle_sso_command
 from .settings import handle_option_command, options_command_list, options_schema_list
 from .tools import NotificationQueue, RequestLoggingTool, TaskManager, \
-    prepare_url_prefix, str_to_bool
+    prepare_url_prefix, str_to_bool, build_url
```
Comment (avanthakkar) [inline src/pybind/mgr/dashboard/module.py:36]: `ERROR: /home/jenkins-build/build/workspace/ceph-pull-requests/src/pybind/mgr/dashboard/module.py Imports are incorrectly sorted and/or formatted.`
```suggestion
    build_url, prepare_url_prefix, str_to_bool
```

File: src/pybind/mgr/dashboard/module.py [hunk 2]
```diff
@@ -191,13 +191,12 @@ def _configure(self):
...
 
-        uri = "{0}://{1}:{2}{3}/".format(
-            'https' if use_ssl else 'http',
-            server_addr,
-            server_port,
-            self.url_prefix
+        base_url = build_url(
+            scheme='https' if use_ssl else 'http',
+            host=server_addr,
+            port=server_port,
```
Comment (avanthakkar) [inline src/pybind/mgr/dashboard/module.py:197]:
```suggestion
            port=server_port
```

**Review by avanthakkar — CHANGES_REQUESTED** [review]
LGTM! Just left some minor changes. Thanks @sebastian-philipp

**Comment by nizamial09** [comment]
jenkins test dashboard

**Comment by sebastian-philipp** [comment]
```
lint run-test-pre: PYTHONHASHSEED='782397422'
lint run-test: commands[0] | flake8
./rest_client.py:331:23: W291 trailing whitespace
./rest_client.py:332:28: W291 trailing whitespace
./rest_client.py:333:28: W291 trailing whitespace

**Review by neha-ojha — APPROVED** [review]
👍

**Review by p-se — APPROVED** [review]
lgtm

**Review by avanthakkar — APPROVED** [review]

**Comment by tchaikov** [comment]
@alfonsomthd in future, could you please add a "Reviewed-by" line in the commit message of the merge commit when merging a PR?

**Comment by alfonsomthd** [comment]
> @alfonsomthd in future, could you please add a "Reviewed-by" line in the commit message of the merge commit when merging a PR?

Of course, sorry about that. I completely forgot about it.

**Pull request merged**